### PR TITLE
add constructor for ScaLAPACKMatrix

### PR DESCRIPTION
**Pull request by BenBrands**
An instance of ScaLAPACKMatrix is filled with content read from disk during construction

### CODE REVIEW

**Comment by davydden** [comment]
i will wait a few days and if no one objects, will merge...

File: include/deal.II/lac/scalapack.h [hunk 1]
```diff
@@ -136,6 +136,20 @@ class ScaLAPACKMatrix : protected TransposeTable<NumberType>
...
+   *
+   * Loads the matrix from file @p filename using HDF5.
+   * In case that deal.II was built without HDF5
+   * a call to this function will cause an exception to be thrown.
+   */
+  ScaLAPACKMatrix(
+    const std::string &                                       filename,
+    const std::shared_ptr<const Utilities::MPI::ProcessGrid> &process_grid,
+    const size_type row_block_size    = 32,
+    const size_type column_block_size = 32);
```
Comment (bangerth) [inline include/deal.II/lac/scalapack.h]: These last two arguments aren't documented. Can you add documentation for them?

Comment (bangerth) [reply include/deal.II/lac/scalapack.h]: That's in fact also the case for the preceding two constructors -- it would be nice to add that there as well.

**Comment by bangerth** [comment]
Let's merge this, since the comments are tangential to what is necessary to get this ready for merge.

@BenBrands -- I would appreciate if you could add the missing documentation of the block sizes for all constructors in a follow-up PR.

**Pull request merged**